### PR TITLE
LibWeb: Implement CSS Transforms on stacking contexts

### PR DESCRIPTION
**Pull request by skyrising**
This also adds `translate()` and `translateX()` support in addition to the previously implemented `translateY()`

### CODE REVIEW

**Comment by awesomekling** [comment]
Hmm hold on, CSS transforms shouldn't really change the layout of the page. It should only affect how and where the affected boxes get painted, right?

**Comment by skyrising** [comment]
True. I kinda just blindly followed what `translateY` was doing. This way everything would break once we set `affects-layout: false` for `transform`. I guess I'll shave some more yaks and do it properly.

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 1]
```diff
@@ -3917,7 +3921,7 @@ RefPtr<StyleValue> Parser::parse_transform_value(Vector<StyleComponentValueRule>
             } else if (value.is(Token::Type::Number)) {
                 auto number = parse_numeric_value(value);
                 values.append(number.release_nonnull());
-            } else {
+            } else if (!value.is(Token::Type::Comma) && !value.is(Token::Type::Whitespace)) {
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:3924]: This logic isn't quite right - it allows things like `this(,,,,,,,,,,)`. CSS requires that we ignore any properties that we don't recognize or that don't fit the grammar, which means we need to catch weird cases like this and return nullptr.

Long term we probably want to make a JSON file of transform-functions and generate some code for checking their arguments, but for now: after successfully appending to `values`, look ahead and make sure that either we get a `Whitespace? Comma Whitespace?` sequence, or that we reach the last value without seeing a comma. That'll mean changing this loop to go by index.

**Comment by awesomekling** [comment]
This looks very cool! There's a small conflict that needs to be resolved before we can merge it, sorry about that.

**Pull request merged**